src/backend/joanie/core/models/contracts.py

```diff
@@ -0,0 +1,232 @@
+"""
+Declare and configure the models for Joanie's contracts
+"""
+import hashlib
+import logging
+import textwrap
+
+from django.conf import settings
+from django.core.exceptions import PermissionDenied
+from django.db import models
+from django.utils.functional import lazy
+from django.utils.translation import gettext_lazy as _
+
+import markdown
+
+from joanie.core import enums
+from joanie.core.utils import image_to_base64
+from joanie.core.utils.issuers import generate_document
+
+from .base import BaseModel
+from .products import Order
+
+logger = logging.getLogger(__name__)
+
+
+class ContractDefinition(BaseModel):
+    """
+    Contract definition describes template and markdown to generate user contracts
+    """
+
+    title = models.CharField(_("title"), max_length=255)
+    description = models.TextField(_("description"), blank=True)
+    body = models.TextField(_("body"), blank=True)
+    language = models.CharField(
+        max_length=10,
+        choices=lazy(lambda: settings.LANGUAGES, tuple)(),
+        verbose_name=_("language"),
+        help_text=_("Language of the contract definition"),
+    )
+
+    template = models.CharField(
+        _("template to generate pdf contract"),
+        max_length=255,
+        blank=True,
+        null=True,
+    )
```

Comment:
Should be a choice field to choose from contract names? should be a required field.
```suggestion
    name = models.CharField(
        _("template name"),
        max_length=255,
        choices=settings.CONTRACT_NAMES,
        default="??"
    )
```